### PR TITLE
En 6049/use sharding.nodes coordinator in boostrap

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: epochStart/bootstrap/syncValidatorStatus.go [hunk 1]
```diff
@@ -56,13 +62,51 @@ func NewSyncValidatorStatus(args ArgsNewSyncValidatorStatus) (*syncValidatorStat
...
+	if err != nil {
+		return nil, err
+	}
+
+	waitingValidators, err := sharding.NodesInfoToValidators(waitingNodesInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	consensusGroupCache, err := lrucache.NewCache(50)
```
Comment (iulianpascalau) [inline epochStart/bootstrap/syncValidatorStatus.go:77]: is this enough? Maybe use a const?

File: process/peer/process.go [hunk 2]
```diff
@@ -167,13 +167,18 @@ func (vs *validatorStatistics) saveUpdatesForNodesMap(
 	nodesMap map[uint32][][]byte,
 	peerType core.PeerType,
 ) error {
-	for shardID, pks := range nodesMap {
-		err := vs.saveUpdatesForList(pks, shardID, peerType)
+	for shardID := uint32(0); shardID < vs.shardCoordinator.NumberOfShards(); shardID++ {
```
Comment (iulianpascalau) [inline process/peer/process.go:170]: please add a TODO here to further investigates what is really going on. Someone else might pick up the task.

File: sharding/indexHashedNodesCoordinator.go [hunk 3]
```diff
@@ -185,8 +185,8 @@ func (ihgs *indexHashedNodesCoordinator) setNodesPerShards(
 		nodesConfig.leavingList = append(nodesConfig.leavingList, validator)
 	}
 
-	nodesList, ok := eligible[core.MetachainShardId]
-	if !ok || len(nodesList) < ihgs.metaConsensusGroupSize {
+	nodesList := eligible[core.MetachainShardId]
```
Comment (iulianpascalau) [inline sharding/indexHashedNodesCoordinator.go:188]: :+1:

File: epochStart/bootstrap/syncValidatorStatus.go (hunk 1 above)
Comment (sasurobert) [reply epochStart/bootstrap/syncValidatorStatus.go:77]: this nodescoordinator is used only to create the setup. not to compute validators group.

Comment (sasurobert) [reply epochStart/bootstrap/syncValidatorStatus.go:77]: added const.

File: process/peer/process.go (hunk 2 above)
Comment (sasurobert) [reply process/peer/process.go:170]: done

File: epochStart/bootstrap/interface.go [hunk 4]
```diff
@@ -1,32 +1,28 @@
...
-		currMetaBlock *block.MetaBlock,
-		prevMetaBlock *block.MetaBlock,
-		publicKey []byte,
-	) (*sharding.NodesCoordinatorRegistry, uint32, error)
+	NodesConfigFromMetaBlock(currMetaBlock *block.MetaBlock, prevMetaBlock *block.MetaBlock) (*sharding.NodesCoordinatorRegistry, uint32, error)
 	IsInterfaceNil() bool
 }
 
-// EpochStartInterceptor -
+// EpochStartInterceptor defines the methods to sync and epoch start metablock
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply epochStart/bootstrap/interface.go:16]: done

File: epochStart/bootstrap/process.go [hunk 5]
```diff
@@ -476,15 +475,13 @@ func (e *epochStartBootstrap) saveSelfShardId() {
 	}
 
 	destShardID := core.MetachainShardId
-	if e.destinationShardAsObserver == "metachain" {
+	if e.destinationShardAsObserver == core.MetachainShardId {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply epochStart/bootstrap/process.go:478]: simplified.

**Pull request merged**